File: 0792-binary-search/0792-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int ans = INT_MAX;int s=0 ; int e = nums.size()-1;
        while(s<=e){
            int mid = s+(e-s)/2;
            if(nums[mid]==target){
                ans = min(ans,mid);
                e=mid-1;
            }
            else if(nums[mid]>target){
                e=mid-1;
            }
            else{
                s=mid+1;
            }
        }
        if(ans==INT_MAX){
            return -1;
        }
        else{
            return ans;
        }
    }
};

File: 0792-binary-search/0792-binary-search.java
class Solution {

    public int rec(int i,int j,int[] nums, int target){
        if(i>j){
            return -1;
        }
        int mid = i + (j-i)/2;
        if(i<=j && target < nums[mid]){
            j=mid-1;
            return rec(i,j,nums,target);
        }
        else if(i<=j && target >nums[mid]){
            i=mid+1;
            return rec(i,j,nums,target);
        }
        else if(i<=j && target == nums[mid]){
            return mid;
        }
        return 0;
    }

    public int search(int[] nums, int target) {
        int i=0;
        int j=nums.length-1;

        int result = rec(i,j,nums,target);

        return result;
    }
}